static/js/user_pill.js

```diff
@@ -94,7 +95,8 @@ export function has_unconverted_data(pill_widget) {
 }
 
 export function typeahead_source(pill_widget) {
-    const persons = people.get_realm_users();
+    let persons = people.get_realm_users();
+    persons = persons.filter((person) => !muting.is_user_muted(person.user_id));
```

Comment:
This will only remove it, recipient typeahead but pills for muted users can still be created by pasting email of users in recipient box. e.g.: `user7@zulipdev.com, user6@zulipdev.com`.
If we want to prevent this then we can modify `create_item_from_email` function in `user_pill.js` to return undefined if the user is muted but this will have some other implications wherever user_pill.js is used for creating user pills.

Reply:
If User A has muted User B then this typeahead source change and above suggested copy-paste change will affect other places too:
* User A cannot add User B to any user group(through UI).
* User A cannot add User B to stream using stream-edit UI.
* User A cannot create pill for User B in recipient box(which means cannot send private message).

Reply:
I don't think we want to totally *prevent* A from adding B in any input, but to just make sure B does not clutter up the typeahead results for A. Maybe I should be writing this in the commit message?

Reply:
Let's make a `people.get_active_unmuted_users` function and call that from all of these code paths -- that's going to avoid duplicating this filter call in N places.

Reply:
On the functionality question, @WesleyAC @alya thoughts on which of "Add subscribers to stream", "Compose box typeahead", and "User groups typeahead" should allow muted users?  I think the policy we'd discussed had been that only "Compose box typeahead" should hide them.

Reply:
The [doc](https://docs.google.com/document/d/1UFQnai5t8R6Fp8lDiWlO9XWuqTUrMrI1tMd-7e7ml3Y/edit#heading=h.vro3nqktoc7j) just says "typeahead lists". Maybe we should update that too?

Reply:
Yeah, I meant "compose typeahead"; we should update that in the doc.

Reply:
Yeah, I think it's nice to have a consistent pattern of showing muted users in the menus, including "Add subscribers to stream" and "User groups typeahead".

Removing muted users from "Compose box typeahead" makes sense to me as well.